src/groupeddatatable/grouping.jl

```diff
@@ -371,12 +369,11 @@ end
 
 function _makeheaders{T<:Function}(fs::Vector{T}, cn::Vector{Symbol})
     fnames = _fnames(fs) # see other/utils.jl
-    reshape([Symbol(colname,'_',fname) for fname in fnames, colname in cn],
-            length(fnames)*length(cn))
+    [Symbol(colname,'_',fname) for fname in fnames for colname in cn]
 end
 
 function _aggregate{T<:Function}(d::AbstractDataTable, fs::Vector{T}, headers::Vector{Symbol}, sort::Bool=false)
-    res = DataTable(colwise(fs, d), headers)
+    res = DataTable(Any[vcat(f(d[i])) for f in fs for i in 1:size(d, 2)], headers)
```

Comment:
https://github.com/JuliaStats/DataFrames.jl/pull/882 only applies to the keyword argument constructor. Using the `DataTable(columns::Array{Any}, colnames::Array{Symbol})` constructor doesn't go through the same functions and each element of `columns` must also be an array. We can change that if you'd like but we'll just end up promoting scalars to Arrays within the constructor in that case, rather than doing it here. We'd also be adding additional checks to the constructor